src/fastalloc/lru.rs

```diff
@@ -0,0 +1,240 @@
+use alloc::vec::Vec;
+use alloc::vec;
+use hashbrown::HashSet;
+use core::{fmt, ops::IndexMut};
+use std::{ops::Index, print};
+use crate::{PReg, PRegSet, RegClass};
+
+use std::{println, format};
+
+/// A least-recently-used cache organized as a linked list based on a vector.
+pub struct Lru {
+    /// The list of node information.
+    ///
+    /// Each node corresponds to a physical register.
+    /// The index of a node is the `address` from the perspective of the linked list.
+    pub data: Vec<LruNode>,
+    /// Index of the most recently used register.
+    pub head: usize,
+    /// Class of registers in the cache.
+    pub regclass: RegClass,
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct LruNode {
+    /// The previous physical register in the list.
+    pub prev: usize,
```

Comment:
These can almost certainly be `u16` and we'll save a little memory (so cache locality) by doing so...

Reply:
Is there any reason why it can't be a `u8`?

Reply:
I had been thinking about the total number of possible `PReg`s and for some reason had thought we could have more than 256 of them across all classes; but (i) we actually can't, since the entire `PReg` is packed into 8 bits, and (ii) your LRU data structure is per-class anyway. So given both those reasons it can definitely be a `u8`!